src/libraries/Common/src/System/Security/Cryptography/DSAKeyFormatHelper.cs

```diff
@@ -105,6 +140,24 @@ internal static class DSAKeyFormatHelper
             ret.Y = y.ExportKeyParameter(ret.P.Length);
         }
 
+        private static bool IsValidPLength(long pBitLength)
+        {
+            return pBitLength switch
+            {
+                // FIPS 186-3/186-4
+                1024 or 2048 or 3072 => true,
+                // FIPS 186-1/186-2
+                >= 512 and < 1024 => pBitLength % 64 == 0,
+                _ => false,
+            };
+        }
+        private static bool IsValidQLength(long qBitLength)
+        {
+            // FIPS 186-1/186-2 only allows 160
+            // FIPS 186-3/186-4 allow 160/224/256
+            return qBitLength is 160 or 224 or 256;
```

Comment:
Are any of these things the kind of values where if the OS were to update to support more we'd be incorrectly blocking values that would otherwise work?

Reply:
FIPS 186-4 defines the possible Q sizes, all of which are handled here. FIPS 186-5 withdrew DSA entirely because it is obsolete and DSA is "done". So the chances of needing to support additional Q lengths are slim to none.

That likewise goes for any other sizes and parameters of DSA. All of the parameters have defined ranges, they are handled correctly here, and new parameter sizes won't happen.